server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateDataStreamService.java

```diff
@@ -394,6 +412,13 @@ private static ClusterState createFailureStoreIndex(
             return currentState;
         }
 
+        var settingsBuilder = Settings.builder().put(IndexMetadata.SETTING_INDEX_HIDDEN, true);
+        // Optionally set a custom refresh interval for the failure store index.
+        var refreshInterval = clusterService.getSettings().getAsTime(FAILURE_STORES_REFRESH_INTERVAL_SETTING_NAME, null);
```

Comment:
Still not really sure whether I want to do `clusterService.getSettings()` here, or do it higher up in the method chain (so a couple of methods accept `Settings` rather than `ClusterService`). Any thoughts on this are welcome.

Reply:
Hm, I think there is a difference on how the updates are received but I cannot remember the difference right now. Is this setting going to be dynamic? If yes, we need to dive into this and better understand the right way of accessing it.

Reply:
Nit: I would also extract it on a little method, this is not the most common way to read a setting, so I think having it as a method might make it a bit more explicit.

Reply:
I would say this is a static setting, as it'll only be set once, just like we do [here](https://github.com/elastic/elasticsearch-serverless/blob/932a6f88426dcba229b2103f337c3f3a1284fc71/modules/serverless-datastream/src/main/java/co/elastic/elasticsearch/serverless/datastream/ServerlessDataStreamPlugin.java#L44).

And for context, [this](https://github.com/nielsbauman/elasticsearch/blob/0de085fe125860f6dd79f3ad9a600c819f3826fb/server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamLifecycle.java#L63-L63) is what I used for reference.

Reply:
But, I think my question is unrelated to whether this is a dynamic or static setting, as I think it's more a question of code styling. If I were suggesting storing the `cluster.getSettings` result in a class field/variable somewhere, I'd agree that it makes a difference (as `Settings` objects are immutable). But here, I'm just questioning at what method/chain level we invoke `cluster.getSettings` (which will only have impact on styling and testing/mocking).

Reply:
Ah clear, thank you for clarifying the question. Personally, I would prefer to pass the settings around and not the cluster service,  for the following reasons:
- It's easier to test, you can just pass the settings instead of the whole cluster service.
- Scope wise, there is no reason to pass the whole cluster service, when you are not looking for more than the settings.
- You could argue then, why would pass the `Settings` and not just the value of the setting we are interested. I think here it makes sense to pass the settings and consider that it's the concern of this method to know how to extract the necessary information, otherwise all callers would need to do this.

What do you think?

Reply:
Yeah I've been trying to remember why I chose to do it the other way (as I was already doubting in the beginning), but can't remember 😅. I agree with what you said, so I've updated it. Thanks!